authors/apps/authentication/serializers.py

```diff
@@ -16,22 +18,91 @@ class RegistrationSerializer(serializers.ModelSerializer):
     password = serializers.CharField(
         max_length=128,
         min_length=8,
-        write_only=True
+        write_only=True,
+    )
+
+    """
+    Ensure that the email entered is unique and give a descriptive
+    error message if a duplicate email is entered
+    """
+    email = serializers.EmailField(
+        validators=[UniqueValidator(
+            User.objects.all(), 'That email is already used. '
+            'Sign in instead or try another')]
+    )
+
+    """
+    Ensure that the username entered is unique and give a descriptive
+    error message if a duplicate username is entered
+    """
+    username = serializers.CharField(
+        validators=[UniqueValidator(
+            User.objects.all(), 'That username is taken. Please try another')]
     )
 
     """
     The client should not be able to send a token along with a registration
     request. Making `token` read-only handles that for us.
     """
-
     class Meta:
+        # RegistrationSerializer uses User model
         model = User
+
         """
         List all of the fields that could possibly be included in a request
         or response, including fields specified explicitly above.
         """
         fields = ['email', 'username', 'password']
 
+    def validate(self, data):
+        """
+        The `validate` method is where we make sure that the current
+        instance of `RegistrationSerializer` has "valid". In the case of registering
+        a user, this means validating that they've provided an email, username
+        and password.
+        """
+        email = data.get('email', None)
```

Comment:
@micahoriaso This can be compressed to a single line variable assignment. i.e a,b = valueA, valueB